### PR TITLE
[JBIDE-26408] warn that debugging nodejs only supports node.js <=7

### PR DESCRIPTION
**Pull request by adietish**
Signed-off-by: Andre Dietisheim <adietish@redhat.com>

# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

**Comment by adietish** [comment]
@jeffmaury please review. 
This is a simple warning dialog when server adapter for Node.js starts in debug mode: we only support Node.js <= v7. A checkbox allows the user not to get bothered any further. Me feels that we need this given that without it, users have no idea why this doesnt work, they have the debugger complaining not to be able to connect to the v8 on OpenShift. Given that we're not able to detect the Node.js that's running, warning upfront seems the only possible solution.
On the technical level, we need to open a dialog in a core plugin where there are no ui packages available. I thus had to create an extension point in org.jboss.tools.openshift.core (that org.jboss.tools.openshift.ui would contribute to) that allows access to simple ui messaging (only #warn for now).

File: plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/IDialogProvider.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Red Hat, Inc.
```
Comment (jeffmaury) [inline plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/IDialogProvider.java]: Should be 2019

File: plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/OpenShiftServerUtils.java [hunk 2]
```diff
@@ -231,6 +231,7 @@ private static void updateServer(IServerWorkingCopy server) {
 		server.setAttribute(IDeployableServer.DEPLOY_DIRECTORY_TYPE, IDeployableServer.DEPLOY_CUSTOM);
 		server.setAttribute(IDeployableServer.ZIP_DEPLOYMENTS_PREF, true);
 		server.setAttribute(Server.PROP_AUTO_PUBLISH_TIME, DEFAULT_AUTO_PUBLISH_DELAY);
+
```
Comment (jeffmaury) [inline plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/OpenShiftServerUtils.java]: Is this change really required ?

File: plugins/org.jboss.tools.openshift.js/src/org/jboss/tools/openshift/js/server/behaviour/OpenShiftNodejsLaunchController.java [hunk 3]
```diff
@@ -16,19 +16,27 @@
...
 	public OpenShiftNodejsLaunchController() {
 		super();
 	}
 
 	@Override
 	protected void startDebugging(OpenShiftServerBehaviour beh, DebugContext context, IProgressMonitor monitor) {
+		// https://issues.jboss.org/browse/JBIDE-26408
+		Dialogs.INSTANCE.warn("Node.js 8+ unsupported	", 
+				"Debugging is only supported up to Node.js 7. If your Node.js is newer, debugging will not work.\n"
+				+ "If using an OpenShift template to create the application, set 'NODEJS_PROPERTY' to '6'",
```
Comment (jeffmaury) [inline plugins/org.jboss.tools.openshift.js/src/org/jboss/tools/openshift/js/server/behaviour/OpenShiftNodejsLaunchController.java]: Why do we recommand Node 6 when the text claims we support up to Node 7 ?

**Comment by adietish** [comment]
@jeffmaury thx for correcting things right away.

File: plugins/org.jboss.tools.openshift.js/src/org/jboss/tools/openshift/js/server/behaviour/OpenShiftNodejsLaunchController.java (hunk 3 above)
Comment (adietish) [reply plugins/org.jboss.tools.openshift.js/src/org/jboss/tools/openshift/js/server/behaviour/OpenShiftNodejsLaunchController.java]: I amended this change to your commit.
